### PR TITLE
Adds alerts

### LINKED ISSUE #12: Add alerts
Create an API to show alerts on the site. The API should support the following types of alerts
- Plain (gray?)
- Info (blue)
- Success (green)
- Warning (warning)
- Error (red)

Check out [react-alert](https://www.npmjs.com/package/react-alert) for a library we can use as inspiration for the API. Note, we don't want to use a library because we want something lightweight and customizable.

### PR DESCRIPTION
**Pull request by swkeever**
Closes #12 

Adds a small library for making alerts. The API and code was heavily inspired by the [react-alert](https://www.npmjs.com/package/react-alert) library.

The API supports 4 different kinds of alerts:
- info
- success
- warning
- error

Each alert type is styled appropriately.

Clients use the alert functionality by using the `useAlert` hook. For example
```javascript
const alert = useAlert()
useEffect(() => alert.show('hello'), [])
```

**You cannot have the alert in the main body of the component function otherwise an infinite loop will occur.** Therefore you must put any alert inside of click handler function or `useEffect`.

### CODE REVIEW

File: src/lib/alerts/use-alert.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * To avoid infinite loops, clients must call this function in a code path
+ * that runs once, not every time the component renders.
+ *
+ * Examples of valid places to do an alert:
+ *   - in a useEffect block with an empty dependency array
+ *   - in an onClick callback from a button click
+ */
+export default function useAlert(): AlertClient {
+  const alert = useContext(AlertContext);
```
Comment (andrew4699) [inline src/lib/alerts/use-alert.ts:16]: Simplify to `return useContext(AlertContext);`

File: src/lib/alerts/use-alert.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useContext } from 'react';
+import AlertContext from './AlertContext';
+import { AlertClient } from './AlertProvider';
+
+/**
+ * Returns a reference to perform alerts.
+ *
+ * To avoid infinite loops, clients must call this function in a code path
```
Comment (andrew4699) [inline src/lib/alerts/use-alert.ts:8]: Aren't React hooks called every time the component renders?

File: package.json [hunk 3]
```diff
@@ -52,7 +53,8 @@
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
     "react-helmet": "^6.1.0",
-    "react-icons": "^3.10.0"
+    "react-icons": "^3.10.0",
+    "react-transition-group": "^4.4.1"
```
[the remark this answers is not in the document]
Comment (swkeever) [reply package.json:57]: :laughing: I'll put that in my backlog

File: src/lib/alerts/AlertContext.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from 'react';
+
+const AlertContext = createContext(undefined);
```
[the remark this answers is not in the document]
Comment (swkeever) [reply src/lib/alerts/AlertContext.ts:3]: TypeScript complains when I don't provide an argument. I changed it to have a more sensible default value (a no-op function)

```
createContext({ show: () => {} })
```

File: src/lib/alerts/AlertProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  AlertOptions, ShowFunction, RemoveFunction, Alert,
+} from './alerts';
+import AlertTemplate from './AlertTemplate';
+
+const defaultOptions: AlertOptions = {
+  duration: undefined,
+  type: 'info',
+};
+
+let uuid = 0;
```
[the remark this answers is not in the document]
Comment (swkeever) [reply src/lib/alerts/AlertProvider.tsx:14]: Using a UNIX timestamp as an ID now

```
        id: Date.now(),
```

**Pull request merged**